File: frontegg-capacitor-plugin/ios/Plugin/FronteggNativePlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(FronteggNativePlugin, "FronteggNative",
           CAP_PLUGIN_METHOD(subscribe, CAPPluginReturnNone);
           CAP_PLUGIN_METHOD(logout, CAPPluginReturnNone);
           CAP_PLUGIN_METHOD(login, CAPPluginReturnNone);
           CAP_PLUGIN_METHOD(getConstants, CAPPluginReturnNone);
)

File: frontegg-capacitor-plugin/ios/Plugin/FronteggNativePlugin.swift
import FronteggSwift
import Foundation
import Combine
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(FronteggNativePlugin)
public class FronteggNativePlugin: CAPPlugin {
    public let fronteggApp = FronteggApp.shared
    var cancellables = Set<AnyCancellable>()

    @objc func getConstants() -> [AnyHashable : Any]! {
      return [
        "baseUrl": fronteggApp.baseUrl,
        "clientId": fronteggApp.clientId,
        "bundleId": Bundle.main.bundleIdentifier!
      ]
    }

    @objc func login() {
        fronteggApp.auth.login()
    }

    @objc func logout() {
      fronteggApp.auth.logout()
    }
    
    
    @objc func subscribe() {
//      fronteggApp.auth.logout()
    }
}
